impl/maven-impl/src/test/java/org/apache/maven/impl/DefaultPluginXmlFactoryTest.java

```diff
@@ -0,0 +1,260 @@
+        assertEquals("1.0.0", descriptor.getVersion());
+    }
+
+    @Test
+    void staticToXmlGeneratesValidXml() {
+        String xml = DefaultPluginXmlFactory.toXml(PluginDescriptor.newBuilder()
+                .name("Sample Plugin")
+                .groupId("org.example")
+                .artifactId("sample-plugin")
+                .version("1.0.0")
+                .build());
+        assertTrue(xml.contains("<name>Sample Plugin</name>"));
+        assertTrue(xml.contains("<groupId>org.example</groupId>"));
+        assertTrue(xml.contains("<artifactId>sample-plugin</artifactId>"));
+        assertTrue(xml.contains("<version>1.0.0</version>"));
+    }
+
+    @Test
+    void writeWithFailingWriterThrowsXmlWriterException() {
+        XmlWriterException exception = assertThrows(
+                XmlWriterException.class,
+                () -> sut.write(XmlWriterRequest.<PluginDescriptor>builder()
+                        .writer(new Writer() {
+                            @Override
+                            public void write(char[] cbuf, int off, int len) {
+                                throw new RuntimeException("Simulated failure");
+                            }
+
+                            @Override
+                            public void flush() {}
+
+                            @Override
+                            public void close() {}
+                        })
+                        .content(PluginDescriptor.newBuilder()
+                                .name("Failing Plugin")
+                                .build())
+                        .build()));
+        assertTrue(exception.getMessage().contains("Unable to write plugin"));
+        assertInstanceOf(RuntimeException.class, exception.getCause());
+    }
+
+    @Test
+    void writeWithNoTargetThrowsIllegalArgumentException() {
+        assertEquals(
+                "writer, outputStream or path must be non null",
+                assertThrows(
+                                IllegalArgumentException.class,
+                                () -> sut.write(XmlWriterRequest.<PluginDescriptor>builder()
+                                        .content(PluginDescriptor.newBuilder()
+                                                .name("No Output Plugin")
+                                                .build())
+                                        .build()))
+                        .getMessage());
+    }
+
+    @Test
+    void readMalformedXmlThrowsXmlReaderException() {
+        XmlReaderException exception = assertThrows(
+                XmlReaderException.class,
+                () -> sut.read(XmlReaderRequest.builder()
+                        .inputStream(new ByteArrayInputStream("<plugin><name>Broken Plugin".getBytes()))
+                        .build()));
+        assertTrue(exception.getMessage().contains("Unable to read plugin"));
+        assertInstanceOf(Exception.class, exception.getCause());
```

Comment:
This assert is almost meaningless

Reply:
`assertInstanceOf(WstxEOFException.class, exception.getCause());`